fontforge/python.c

```diff
@@ -2391,33 +2425,31 @@ static Py_ssize_t PyFFContour_Length( PyObject *self ) {
 return( ((PyFF_Contour *) self)->pt_cnt );
 }
 
+static PyFF_Contour *PyFFPointList_Parse(PyObject *args);
+
 static PyObject *PyFFContour_Concat( PyObject *_c1, PyObject *_c2 ) {
-    PyFF_Contour *c1 = (PyFF_Contour *) _c1, *c2 = (PyFF_Contour *) _c2;
+    PyFF_Contour *c1 = (PyFF_Contour *) _c1, *c2 = (PyFF_Contour *)_c2;
     PyFF_Contour *self;
     int i;
     PyFF_Contour dummy;
     PyFF_Point *dummies[1];
     double x,y;
 
-    if ( PyType_IsSubtype(&PyFF_PointType, Py_TYPE(c2)) ) {
+    if ( PyType_IsSubtype(&PyFF_ContourType, Py_TYPE(c1)) &&
+         PyType_IsSubtype(&PyFF_ContourType, Py_TYPE(c2)) ) {
```

Comment:
I don't immediately understand what is going on here. Does ContourType inherit from PointType?

Reply:
Oh. I see. You reordered it.

Reply:
Any particular reason?

Reply:
I wanted to make sure I catch the double-contour case first because this function now calls `PyFFPointList_Parse()` and vice versa. That's OK because `PyFFContour_Concat()` never calls  `PyFFPointList_Parse()` when passed two contours, and the latter only calls the former with two contours. I wanted to lessen the chance that some future update leads to an infinite call chain.

Reply:
Oh, slightly correction: `PyFFPointList_Parse()` calls `PyFFContour_InPlaceConcat()` and vice versa. I reordered `PyFFContour_Concat()` to match `PyFFContour_InPlaceConcat()`.